drivers/counter/counter_rpi_pico_pit.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2024 Navimatix GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <zephyr/drivers/counter.h>
+#include <zephyr/sys/slist.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct rpi_pico_pit_callback {
+	/** This is meant to be used in the driver and the user should not
+	 * mess with it
+	 */
+	sys_snode_t node;
+
+	/** Actual callback function being called when relevant. */
+	counter_top_callback_t callback;
+
+	/** User data of the callback function */
+	void *top_user_data;
+
+	/** Slice number of the pit channel the callback belongs to */
+	uint32_t slice;
+};
+
+/**
+ * @brief Function to get whether the PIT channel has a pending interrupt
```

Comment:
Same for other functions as well
```suggestion
 * @brief Get whether the PIT channel has a pending interrupt
```